lib/jekyll/utils.rb

```diff
@@ -34,6 +34,7 @@ def deep_merge_hashes(master_hash, other_hash)
     # Thanks to whoever made it.
     def deep_merge_hashes!(target, overwrite)
       overwrite.each_key do |key|
+        next if overwrite[key].nil? && target.key?(key)
```

Comment:
> This makes it impossible to nil-ify a key :(

The problem is that the drop will *always* have a `page` key, so what is the other option?